core/sql/sqlcomp/nadefaults.cpp

```diff
@@ -4333,59 +4336,8 @@ void NADefaults::updateSystemParameters(NABoolean reInit)
 
     case MAX_ESPS_PER_CPU_PER_OP:
       {
-        // set 2 ESPs per node, as a starting point.
-        #define DEFAULT_ESPS_PER_NODE 2
-
-        Lng32 numESPsPerNode = DEFAULT_ESPS_PER_NODE;
-        Lng32 coresPerNode = 1;        
-          // Make sure the gpClusterInfo points at an NAClusterLinux object.
-          // In osim simulation mode, the pointer can point at a NAClusterNSK
-          // object, for which the method numTSEsForPOS() is not defined.
-        NAClusterInfoLinux* gpLinux = dynamic_cast<NAClusterInfoLinux*>(gpClusterInfo);
-
-          // number of POS TSE
-        Lng32 numTSEsPerCluster = gpLinux->numTSEsForPOS();
-
-          // cluster nodes
-        Lng32 nodesdPerCluster = gpClusterInfo->getTotalNumberOfCPUs();
-
-          // TSEs per node
-        Lng32 TSEsPerNode = numTSEsPerCluster/nodesdPerCluster;
-
-          // cores per node
-        coresPerNode = gpClusterInfo->numberOfCpusPerSMP();
-
-          // For Linux/nt, we conservatively allocate ESPs per node as follows
-          // - 1 ESP per 2 cpu cores if cores are equal or less than TSEs
-          // - 1 ESP per TSE if number of cores is more than double the TSEs
-          // - 1 ESP per 2 TSEs if cores are more than TSEs but less than double the TSEs
-          // - 1 ESP per node. Only possible on NT or workstations
-          //      - number of cores less than TSEs and there are 1 or 2 cpur cores per node
-          //      - number of TSEs is less than cpu cores and there 1 or 2 TSEs per node.
-          //        This case is probable if virtual nodes are used
-
-          // TSEsPerNode is 0 for arkcmps started by the seapilot universal comsumers
-          // in this case we only consider cpu cores
-        if ((coresPerNode <= TSEsPerNode) || (TSEsPerNode == 0))
-        {
-            if (coresPerNode > 1)
-                numESPsPerNode = DEFAULT_ESPS_PER_NODE; 
-        }
-        else if (coresPerNode > (TSEsPerNode*2))
-        {
-             numESPsPerNode = TSEsPerNode;
-        }
-        else if (TSEsPerNode > 1)
-        {
-             numESPsPerNode = TSEsPerNode/2;
-        }
-        else // not really needed since numESPsPerNode is set to 1 from above
-        {
-             numESPsPerNode = DEFAULT_ESPS_PER_NODE;
-        }
-        
-
-        ftoa_((float)(numESPsPerNode)/(float)(coresPerNode), valuestr);
+        float espsPerCore = computeNumESPsPerCore(FALSE);
```

Comment:
I see.  I think one section of code was missing. My mistake.

Please review the updated version. Sorry.



On Thu, Jan 7, 2016 at 12:53 PM, DaveBirdsall <notifications@github.com>
wrote:

> In core/sql/sqlcomp/nadefaults.cpp
> <https://github.com/apache/incubator-trafodion/pull/250#discussion_r49110123>
> :
>
> > -        else if (coresPerNode > (TSEsPerNode*2))
> > -        {
> > -             numESPsPerNode = TSEsPerNode;
> > -        }
> > -        else if (TSEsPerNode > 1)
> > -        {
> > -             numESPsPerNode = TSEsPerNode/2;
> > -        }
> > -        else // not really needed since numESPsPerNode is set to 1 from above
> > -        {
> > -             numESPsPerNode = DEFAULT_ESPS_PER_NODE;
> > -        }
> > -
> > -
> > -        ftoa_((float)(numESPsPerNode)/(float)(coresPerNode), valuestr);
> > +        float espsPerCore = computeNumESPsPerCore(FALSE);
>
> Not sure I understand. We pass FALSE here to computeNumESPsPerCore, and
> when I read that method, it goes to the else case (that is, !aggressive).
> So it appears "not aggressive" is currently hard-coded?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/250/files#r49110123>.
>



-- 
Regards, --Qifan